datashader/datashape/coretypes.py

```diff
@@ -376,10 +372,10 @@ class String(Unit):
     def __init__(self, *args):
         if len(args) == 0:
             fixlen, encoding = None, None
-        if len(args) == 1:
-            if isinstance(args[0], _strtypes):
+        elif len(args) == 1:
+            if isinstance(args[0], str):
                 fixlen, encoding = None, args[0]
-            if isinstance(args[0], _inttypes):
+            if isinstance(args[0], int):
                 fixlen, encoding = args[0], None
         if len(args) == 2:
```

Comment:
```suggestion
        elif len(args) == 2:
```

To align with changes above.